code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -98,11 +124,295 @@
 		ticker.mode.update_necro_icons_added(Controller)
 		ticker.mode.update_all_necro_icons()
 		ticker.mode.risen.Add(Controller)
-	name += " ([rand(1,1000)])"
 
-/mob/living/simple_animal/hostile/necro/copy/ListTargets()
-	. = ..()
-	return . - creator
+	if(name == "zombie" || name == "skeleton")
```

Comment:
Actually best case scenario would be `if (name == initial(name) && !unique_name)` or something like that, so on-map objects don't get screwed over.